controller/api/public/prometheus.go

```diff
@@ -91,7 +91,7 @@ func promDstGroupByLabelNames(resource *pb.Resource) model.LabelNames {
 func promQueryLabels(resource *pb.Resource) model.LabelSet {
 	set := model.LabelSet{}
 	if resource != nil {
-		if resource.Name != "" && resource.GetType() != k8s.Service {
+		if resource.Name != "" && resource.GetType() != k8s.Service && resource.GetType() != k8s.TrafficSplit {
```

Comment:
If we don't call this function on a traffic-split resource (as I suggest below) then I don't think we need this change.